### PR TITLE
JS API and an add-on to consume it.

### PR DESCRIPTION
**Pull request by rnewman**

### CODE REVIEW

**Comment by rnewman** [comment]
I'm addressing some review comments, which led me to find an error in the project file — we can't include node-only sources in the top-level project source paths. I'll wrap all this up tomorrow.

File: addon/build.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules/.bin/webpack -p
```
Comment (ncalexan) [inline addon/build.sh]: I don't understand how the actual `datomish.js` file gets into the addon.  Can you clarify this?

File: addon/src/index.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+var self = require("sdk/self");
```
Comment (ncalexan) [inline addon/src/index.js]: License?  PD?

File: project.clj [hunk 3]
```diff
@@ -12,17 +12,24 @@
...
+                 ;; Instead, add the :node profile:
+                 ;;   lein with-profile node install
+                 ;"src/node"
+                 ]
 
   :cljsbuild {:builds
               {
                :release-node
                {
-                :source-paths   ["src/node" "src/common"]
```
Comment (ncalexan) [inline project.clj]: I needed to roll this back -- see later.

File: src/browser/datomish/cljify.cljs [hunk 4]
```diff
@@ -0,0 +1,47 @@
+(ns datomish.cljify)
```
Comment (ncalexan) [inline src/browser/datomish/cljify.cljs]: Pretty sure there's trailing whitespace in this file, possibly others.

File: addon/build.sh (hunk 1 above)
Comment (rnewman) [inline addon/build.sh]: I just pushed a commit to add a `cp`.

File: addon/src/index.js (hunk 2 above)
Comment (rnewman) [inline addon/src/index.js]: Done.

File: src/browser/datomish/cljify.cljs (hunk 4 above)
Comment (rnewman) [inline src/browser/datomish/cljify.cljs]: Done.

**Comment by rnewman** [comment]
Landed.

**Pull request closed without merging**